From: Phil Elwell <phil@raspberrypi.com>
Date: Fri, 7 Feb 2020 09:51:31 +0000
Subject: overlays: Add timeout_ms parameter to gpio-poweroff

The timeout_ms parameter specifies in milliseconds how long the kernel
waits for power-down before issuing a WARN. The default value is 3000 ms.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/overlays/README                    | 2 ++
 arch/arm/boot/dts/overlays/gpio-poweroff-overlay.dts | 1 +
 2 files changed, 3 insertions(+)

diff --git a/arch/arm/boot/dts/overlays/README b/arch/arm/boot/dts/overlays/README
index ac1f69c9e4e2..572edc7818ea 100644
--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -821,6 +821,8 @@ Params: gpiopin                 GPIO for signalling (default 26)
         input                   Set if the gpio pin should be configured as
                                 an input.
         export                  Set to export the configured pin to sysfs
+        timeout_ms              Specify (in ms) how long the kernel waits for
+                                power-down before issuing a WARN (default 3000).
 
 
 Name:   gpio-shutdown
diff --git a/arch/arm/boot/dts/overlays/gpio-poweroff-overlay.dts b/arch/arm/boot/dts/overlays/gpio-poweroff-overlay.dts
index bb8cd3bf264d..416aa2bc797a 100644
--- a/arch/arm/boot/dts/overlays/gpio-poweroff-overlay.dts
+++ b/arch/arm/boot/dts/overlays/gpio-poweroff-overlay.dts
@@ -32,5 +32,6 @@
 		active_low =    <&power_ctrl>,"gpios:8";
 		input =         <&power_ctrl>,"input?";
 		export =        <&power_ctrl>,"export?";
+		timeout_ms =    <&power_ctrl>,"timeout-ms:0";
 	};
 };
